### PR TITLE
Deployment badge reflects endpoint liveness (#27)

### LINKED ISSUE #27: Deployment shows planted when endpoint is stale/offline
A deployment keeps showing green "planted" on the dashboard even after its endpoint has gone stale/inactive. The bait may no longer be watched, but the UI still implies healthy coverage. Should visually reflect endpoint liveness - e.g. muted/amber badge when the agent is offline.

### PR DESCRIPTION
**Pull request by LiorFink00**
Closes #27. A deployment kept showing green 'planted' even when its endpoint went stale/offline. `DeploymentOut` now carries `endpoint_status`; `DeployBadge` mutes a planted bait to amber 'planted · offline' when the endpoint isn't online. Backend TDD (online vs stale) + UI build.

### CODE REVIEW

File: tests/test_deployment_liveness.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+def test_planted_on_online_endpoint(client_db):
+    tc, db = client_db
+    tid = _seed(db, _ago(minutes=2))
+    assert _endpoint_status_in_detail(tc, tid) == "online"
+
+
+def test_planted_on_stale_endpoint(client_db):
+    tc, db = client_db
+    tid = _seed(db, _ago(hours=1))
+    assert _endpoint_status_in_detail(tc, tid) == "stale"
```
Comment (avivhalfon11) [inline tests/test_deployment_liveness.py:65]: Tests cover online/stale, but the two genuinely new branches in _deployment_out aren't exercised: endpoint is None -> "inactive", and decommission_requested_at -> "decommissioning". Those are the edge cases most likely to regress later - worth a _seed variant with no endpoint and one with a decommission timestamp.

File: server/thumper/api/routes.py [hunk 2]
```diff
@@ -99,13 +99,20 @@ def _tripwire_out(db, tripwire, *, deployed_count=None, triggered_count=None) ->
 
 def _deployment_out(db, deployment) -> DeploymentOut:
     endpoint = store.get_endpoint(db, deployment.endpoint_id)
+    if endpoint is None:
```
Comment (avivhalfon11) [inline server/thumper/api/routes.py]: Optional: this "decommissioning vs _endpoint_status(last_seen)" rule is now duplicated with _endpoint_out. Consistent today, but consistency-by-copy - if a new status is ever added, both sites have to change in lockstep. A shared helper (e.g. _endpoint_status_full(endpoint)) would make drift impossible.

**Review by avivhalfon11 — APPROVED** [review]
LGTM - and it stays consistent with the existing _endpoint_out status logic, which is the main thing. Left one test-coverage comment and one optional DRY note, nothing blocking.

File: tests/test_deployment_liveness.py (hunk 1 above)
Comment (LiorFink00) [reply tests/test_deployment_liveness.py:65]: Added both: `test_planted_on_decommissioning_endpoint` (fresh last_seen + decommission timestamp -> "decommissioning", proving the override) and `test_planted_on_missing_endpoint` (no endpoint row -> "inactive", not a 500). `_seed` grew a `decommission_requested_at` arg and a `with_endpoint=False` variant.

File: server/thumper/api/routes.py (hunk 2 above)
Comment (LiorFink00) [reply server/thumper/api/routes.py]: Done — extracted `_endpoint_status_full(endpoint)` as the single source of truth (handles missing-endpoint, decommission override, then last_seen). Both `_endpoint_out` and `_deployment_out` call it now, so a new status only has to be added in one place.
